python-checks/src/main/java/org/sonar/python/checks/hotspots/ClearTextProtocolsCheck.java

```diff
@@ -89,28 +85,26 @@ public void initialize(Context context) {
 
     context.registerSyntaxNodeConsumer(Tree.Kind.QUALIFIED_EXPR, ClearTextProtocolsCheck::checkServerCallFromSuper);
 
-    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, ClearTextProtocolsCheck::checkServerBindOverrides);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ClearTextProtocolsCheck::checkServerBindCalls);
+
     new ClearTextProtocolsCheckPart().initialize(context);
   }
 
   private static void checkServerCallFromSuper(SubscriptionContext ctx) {
     QualifiedExpression qualifiedExpression = (QualifiedExpression) ctx.syntaxNode();
     Optional.of(qualifiedExpression)
-      .filter(qe -> isName("serve_forever", qe.name()) && isCallToSensitiveSuperClass(qe))
+      .filter(qe -> SENSITIVE_HTTP_SERVER_METHOD_NAMES.contains(qe.name().name()) && isCallToSensitiveSuperClass(qe))
       .map(qe -> TreeUtils.firstAncestorOfKind(qe, Tree.Kind.CALL_EXPR))
       .flatMap(TreeUtils.toOptionalInstanceOfMapper(CallExpression.class))
       .ifPresent(ce -> ctx.addIssue(ce, message("http")));
   }
 
-  private static void checkServerBindOverrides(SubscriptionContext ctx) {
-    FunctionDef funcDef = (FunctionDef) ctx.syntaxNode();
-    Optional.of(funcDef)
-      .filter(fd -> isName("server_bind", fd.name()) && isParentClassExtendingSensitiveClass(funcDef))
-      .ifPresent(fd -> ctx.addIssue(fd.defKeyword(), fd.rightPar(), message("http")));
-  }
-
-  private static boolean isName(String nameToCheck, Name name) {
-    return nameToCheck.equals(name.name());
+  private static void checkServerBindCalls(SubscriptionContext ctx) {
+    CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+    Optional.ofNullable(callExpression.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
+      .filter(fqn -> SENSITIVE_HTTP_SERVER_BIND_FQN.equals(fqn) && isParentClassExtendingSensitiveClass(callExpression))
```

Comment:
Here I would suggest just having:
```java
.filter(SENSITIVE_HTTP_SERVER_BIND_FQN::equals)
```
and have an if statement at the beginning of the method such as:
```java
if (!isParentClassExtendingSensitiveClass(callExpression)) {
  return;
}
```

Reply:
I would rather check first if the FQN matches. The check of the parent class can be costly as it traverses the graph. But I could modify it to have two filters.